### ISSUE TITLE
Gamepipe.exe has stopped working

### ISSUE THREAD

**Issue by dcjamison** [DjScribbles/GamePipe#4]
I get the following when trying to run the program on my computer:
Gamepipe.exe has stopped working

Problem signature:
  Problem Event Name:   CLR20r3
  Problem Signature 01: GamePipe.exe
  Problem Signature 02: 0.1.0.0
  Problem Signature 03: 56d74bc7
  Problem Signature 04: PresentationCore
  Problem Signature 05: 4.0.30319.34209
  Problem Signature 06: 5348b56b
  Problem Signature 07: 3964
  Problem Signature 08: 19
  Problem Signature 09: System.ArgumentException
  OS Version:   6.1.7601.2.1.0.256.1
  Locale ID:    1033
  Additional Information 1: 0a9e
  Additional Information 2: 0a9e372d3b4ad19135b953a78882e789
  Additional Information 3: 0a9e
  Additional Information 4: 0a9e372d3b4ad19135b953a78882e789

Read our privacy statement online:
  http://go.microsoft.com/fwlink/?linkid=104288&clcid=0x0409

If the online privacy statement is not available, please read our privacy statement offline:
  C:\Windows\system32\en-US\erofflps.txt

**Comment by DjScribbles** [comment]
could you post the contents of the log file usually located at "C:\Program Files (x86)\GamePipe\logs\log.txt". I've gotten a few reports of this, but haven't gotten anyone with a log yet.

**Comment by CMDRBULB** [comment]
I have the same problem. Here is a copy of the log file. Let me know if you have a better way to share these logs than a huge paste dump in the comments.

2016-03-07 16:07:29,586 INFO - -----------------------GamePipe started 0.1.4.27707-----------
2016-03-07 16:07:31,628 ERROR - An unhandled dispatcher exception occurred:
System.ArgumentException: Value does not fall within the expected range. ---> System.ArgumentException: Value does not fall within the expected range.
   --- End of inner exception stack trace ---
   at System.Windows.Media.ColorContext.GetColorContextsHelper(GetColorContextsDelegate getColorContexts)
   at System.Windows.Media.Imaging.BitmapFrameDecode.get_ColorContexts()
   at System.Windows.Media.Imaging.BitmapImage.FinalizeCreation()
   at System.Windows.Media.Imaging.BitmapImage.OnDownloadCompleted(Object sender, EventArgs e)
   at System.Windows.Media.UniqueEventHelper.InvokeEvents(Object sender, EventArgs args)
   at System.Windows.Media.Imaging.LateBoundBitmapDecoder.DownloadCallback(Object arg)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
2016-03-07 16:07:31,653 ERROR - An unhandled exception occurred:
System.ArgumentException: Value does not fall within the expected range. ---> System.ArgumentException: Value does not fall within the expected range.
   --- End of inner exception stack trace ---
   at System.Windows.Media.ColorContext.GetColorContextsHelper(GetColorContextsDelegate getColorContexts)
   at System.Windows.Media.Imaging.BitmapFrameDecode.get_ColorContexts()
   at System.Windows.Media.Imaging.BitmapImage.FinalizeCreation()
   at System.Windows.Media.Imaging.BitmapImage.OnDownloadCompleted(Object sender, EventArgs e)
   at System.Windows.Media.UniqueEventHelper.InvokeEvents(Object sender, EventArgs args)
   at System.Windows.Media.Imaging.LateBoundBitmapDecoder.DownloadCallback(Object arg)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run(Window window)
   at GamePipe.App.Main()
2016-03-07 16:09:09,553 INFO - -----------------------GamePipe started 0.1.4.27707-----------
2016-03-07 16:09:10,446 ERROR - An unhandled dispatcher exception occurred:
System.ArgumentException: Value does not fall within the expected range. ---> System.ArgumentException: Value does not fall within the expected range.
   --- End of inner exception stack trace ---
   at System.Windows.Media.ColorContext.GetColorContextsHelper(GetColorContextsDelegate getColorContexts)
   at System.Windows.Media.Imaging.BitmapFrameDecode.get_ColorContexts()
   at System.Windows.Media.Imaging.BitmapImage.FinalizeCreation()
   at System.Windows.Media.Imaging.BitmapImage.OnDownloadCompleted(Object sender, EventArgs e)
   at System.Windows.Media.UniqueEventHelper.InvokeEvents(Object sender, EventArgs args)
   at System.Windows.Media.Imaging.LateBoundBitmapDecoder.DownloadCallback(Object arg)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
2016-03-07 16:09:10,448 ERROR - An unhandled exception occurred:
System.ArgumentException: Value does not fall within the expected range. ---> System.ArgumentException: Value does not fall within the expected range.
   --- End of inner exception stack trace ---
   at System.Windows.Media.ColorContext.GetColorContextsHelper(GetColorContextsDelegate getColorContexts)
   at System.Windows.Media.Imaging.BitmapFrameDecode.get_ColorContexts()
   at System.Windows.Media.Imaging.BitmapImage.FinalizeCreation()
   at System.Windows.Media.Imaging.BitmapImage.OnDownloadCompleted(Object sender, EventArgs e)
   at System.Windows.Media.UniqueEventHelper.InvokeEvents(Object sender, EventArgs args)
   at System.Windows.Media.Imaging.LateBoundBitmapDecoder.DownloadCallback(Object arg)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run(Window window)
   at GamePipe.App.Main()
2016-03-07 16:09:50,914 INFO - -----------------------GamePipe started 0.1.3.25036-----------
2016-03-07 16:09:51,745 ERROR - An unhandled dispatcher exception occurred:
System.ArgumentException: Value does not fall within the expected range. ---> System.ArgumentException: Value does not fall within the expected range.
   --- End of inner exception stack trace ---
   at System.Windows.Media.ColorContext.GetColorContextsHelper(GetColorContextsDelegate getColorContexts)
   at System.Windows.Media.Imaging.BitmapFrameDecode.get_ColorContexts()
   at System.Windows.Media.Imaging.BitmapImage.FinalizeCreation()
   at System.Windows.Media.Imaging.BitmapImage.OnDownloadCompleted(Object sender, EventArgs e)
   at System.Windows.Media.UniqueEventHelper.InvokeEvents(Object sender, EventArgs args)
   at System.Windows.Media.Imaging.LateBoundBitmapDecoder.DownloadCallback(Object arg)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
2016-03-07 16:09:51,747 ERROR - An unhandled exception occurred:
System.ArgumentException: Value does not fall within the expected range. ---> System.ArgumentException: Value does not fall within the expected range.
   --- End of inner exception stack trace ---
   at System.Windows.Media.ColorContext.GetColorContextsHelper(GetColorContextsDelegate getColorContexts)
   at System.Windows.Media.Imaging.BitmapFrameDecode.get_ColorContexts()
   at System.Windows.Media.Imaging.BitmapImage.FinalizeCreation()
   at System.Windows.Media.Imaging.BitmapImage.OnDownloadCompleted(Object sender, EventArgs e)
   at System.Windows.Media.UniqueEventHelper.InvokeEvents(Object sender, EventArgs args)
   at System.Windows.Media.Imaging.LateBoundBitmapDecoder.DownloadCallback(Object arg)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run(Window window)
   at GamePipe.App.Main()

**Comment by DjScribbles** [comment]
Thanks, this is in line with what I have seen reported on my Greenlight page.

**Comment by DjScribbles** [comment]
I just pushed a commit that may fix this issue, but I don't have a good way to test it myself.
If you guys have VS, do let me know if this resolves your issue, otherwise it will show up in the next release and can be tested then.

This post describes what I believe to be the issue: http://www.hanselman.com/blog/DealingWithImagesWithBadMetadataCorruptedColorProfilesInWPF.aspx

**Comment by DjScribbles** [comment]
I just released the potential fix for this issue in 0.1.5: https://github.com/DjScribbles/GamePipe/releases/tag/0.1.5

@CMDRBULB @dcjamison Can either of you test the release to see if the issue is resolved?

**Comment by DjScribbles** [comment]
Reports from Steam indicate the issue isn't resolved by 0.1.5; so I've uploaded a handful of builds trying different things. There's a batch file to run each build one at a time, and they're numbered 0-4.

Ideally, I'd really appreciate if someone experiencing this issue could download [ImageExceptionTests2.zip](https://github.com/DjScribbles/GamePipe/files/164377/ImageExceptionTests2.zip) and run the launchAllTests.bat. Record a quick note on whether it ran, and whether there were images present.

After the tests, if the folder could be zipped back up (which will include the logs of your running it) and uploaded/linked here, so I can look over the results.

**Comment by CMDRBULB** [comment]
Ok, I ran the bat, both 1 and 2 ran successfully. 1 had images while 2 did not. 0, 3, and 4 all crashed immediately .
Github refuses to allow me to upload the zip, so I sent you a pm on Reddit with a link for the file.

**Comment by SkippySqueeze** [comment]
1 and 2 both work (second and third). 0, 3, and 4 still crash. 1 has images like it should, and 2 does not like it shouldn't.
[ImageExceptionTests2.zip](https://github.com/DjScribbles/GamePipe/files/164600/ImageExceptionTests2.zip)

**Comment by DjScribbles** [comment]
Thanks so much both of you for all your help in trying to get to the bottom of this issue. I have one more zip here to test: [5_UpdatedCacheImage.zip](https://github.com/DjScribbles/GamePipe/files/164610/5_UpdatedCacheImage.zip)

Edit: One more to try, if one of these two doesn't work, I'll be at a loss: 
[6_LocalImgCache.zip](https://github.com/DjScribbles/GamePipe/files/164669/6_LocalImgCache.zip)

**Comment by descention** [comment]
V 0.1.5.910 failed to run for me. 5 and 6 above both worked for me.

**Comment by SkippySqueeze** [comment]
Doesn't crash, has images. Nice job!

**Comment by DjScribbles** [comment]
Excellent. Thanks for everyone who tested this out. This fix should make it into the next release, and I've got a pull request out to the creator of [CachedImage](https://github.com/floydpink/CachedImage) to add the option that was needed to resolve the crash (it wasn't a bug in CachedImage, it just lacked the configuration option needed to suppress errors from corrupt images).

What I find odd is that I was able to load the images for all the games of @SkippySqueeze and another steam user who reported the issue, and didn't encounter a problem. The issue must be localized to whatever Steam content delivery server is serving you data. Maybe it's a regional thing?
